### PR TITLE
Conv2dGrowingModule implementation of natural gradient

### PR DESCRIPTION
**Pull request by TheoRudkiewicz**
For now it only has the computation of the optimal update (natural gradient).

### CODE REVIEW

**Comment by TheoRudkiewicz** [comment]
I split the development in multiple PR, here is the first one !

File: tests/test_conv2d_growing_module.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        self.demo_b = Conv2dGrowingModule(
+            in_channels=2, out_channels=7, kernel_size=3, padding=1, use_bias=True
+        )
+        self.demo_b.layer = self.demo_layer_b
+
+        torch.manual_seed(0)
+        self.input_x = torch.randn(5, 2, 10, 10, device=global_device())
+
+    def test_init(self):
+        # no bias
```
Comment (sylvchev) [inline tests/test_conv2d_growing_module.py:32]: We could use subTest, I'll propose something

**Review by sylvchev — APPROVED** [review]
Ok, we could test einsum and use subTest later, but we need to open issues to keep track.

File: src/gromo/conv2d_growing_module.py [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+from gromo.tensor_statistic import TensorStatistic
+from gromo.tools import compute_mask_tensor_t
+from gromo.utils.utils import global_device
+
+
+class Conv2dAdditionGrowingModule(AdditionGrowingModule):
+    pass
+
+
+class Conv2dGrowingModule(GrowingModule):
```
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/conv2d_growing_module.py:16]: Yes definitely I will add it on next PR !

File: src/gromo/conv2d_growing_module.py [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+            number of samples used to compute the update
+        """
+        assert (
+            self.store_input
+        ), f"The input must be stored to compute the update of S. (error in {self.name})"
+        assert (
+            self.input is not None
+        ), f"The input must be stored to compute the update of S. (error in {self.name})"
+        return (
+            torch.einsum(
```
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/conv2d_growing_module.py:180]: This approximately the same in this case. With tricky operation we could gain ~10% but einsum is a good naive start.

**Pull request merged**